src/hooks/useField.js

```diff
@@ -283,16 +283,19 @@ export default function useField(
 
   // When the form gets dirty and when the value changes, run the validation
   React.useEffect(() => {
-    if (!validatePristine && !meta.isTouched) {
-      return
-    }
+    setTimeout(()=> {
+      const isTouched = meta.isTouched || formApiRef.current.getFieldMeta(fieldName)?.isTouched
```

Comment:
Check both the prop and `formApiRef.current.getFieldMeta(fieldName)?.isTouched` because there may or may not be a race